### ISSUE TITLE
Custom Dropdown Menu Z-Index

### ISSUE THREAD

**Issue by Dizolivemint** [moodlerooms/moodle-theme_snap#41]
The problem with dropdown menus is they have to have a higher z-index than the Moodle Mr. Nav menu in order for that part of the dropdown to be visible. Setting the z-index to 4 allows the dropdown to show, but it also shows up when the user clicks on the Mr. Nav menu. Trying to figure out a way to set the z-index when Mr. Nav opens.

Set the custom navigation menus z-index to 4. Then do something like this using Javascript and jQuery:
var element = document.getElementByID('js-menu-trigger'),
style = window.getComputedStyle(element),
display = style.getPropertyValue('display');
                                    
if(display == 'none') {
    $("#mr-nav").css("z-index", "10")
}
else {
    $("#mr-nav").css("z-index", "3")
}

How would I add this to the Snap template?

**Comment by stuartlamour** [comment]
Thanks @Dizolivemint,

we use the core moodle moodle-has-zindex class where possible which dynamically sets the z-index but may not work well with bootstrap.

If you could let us know a little more about what version of moodle you are using and the dropdown code you are using we can take a look.

Thanks again for reporting,
Stuart

**Comment by Dizolivemint** [comment]
Thank you for your reply @stuartlamour,

Currently using Moodle ver 3.1. 

After reading you post I deleted all of the code from Appearance > Additional HTML > "Within HEAD field" & "When BODY is opened field".

Adding `<div id="header1" style="height: 50px; background-color:blue"></div>` causes `header#mr-nav` to overlap `div#inst5` when its class changes to `.state-visible` because the `a#admin-menu-trigger` is clicked.

This seems due partially to `header#mr-nav` not being at the top of the page anymore.

Setting the z-index higher for the menu and moving the admin-menu-trigger to the left was fixed the issue:
`header#mr-nav { z-index: 3; position: relative; }
 div#inst5 { z-index: 4; }
a#admin-menu-trigger.pull-right { float: left !important }
header#mr-nav.headroom--unpinned { -webkit-animation-name: none !important; animation-name: none !important; }`

Added a second div beneath the first with a menu list in `div#menu-content`. Set its `z-index: 2`. Now the menu is functioning normally and isn't being placed overlapping the `nav#primary-nav` anymore.

Now when `a#js-personal-menu-trigger` is clicked, `nav#primary-nav` displays highest in the stack again. In my original post, the problem was that `nav#primary-nav` was displaying underneath the header. Rewriting the code in the Additional HTML page of Moodle somehow fixed the original issue.

Moving a#admin-menu-trigger to the left and stacking it behind `div#inst5` isn't a solution for mobile because the menu covers the entire width of the screen, covering the menu trigger to close `div#inst5`. We don't care about the admins UX, but other developers may need a better solution.

Thank you for your support,
Miles
